### ISSUE TITLE
As a user i would like to have command line parameters verified and get an detailled error message on start

### ISSUE THREAD

**Issue by legacycode** [iotaledger/iri#1252]
### Description
IRI is using JCommander to parse command line parameters. JCommander has a lot of features in validating the user input by creating converters and parser. It is also possible to use a configuration file with JCommander using the @syntax: http://jcommander.org/#__syntax

Currently there are two implementations in reading IRI config file. One by parsing command line parameter with JCommander and parsing a configuration file by Jackson. I would like to drop the Jackson support and the related maven dependency and focus on JCommander for better command line parameter checks and type based parsing.

What is your opinion about this idea?

### Motivation
Validating command line parameter is a pain at the moment. There is also no user feedback, if wrong parameters for example REMOTE_LIMIT_API are passed.

### Requirements
Getting error messages when passing wrong command line parameters.

### Am I planning to do it myself with a PR?
Yes.

**Comment by GalRogozinski** [comment]
> Validating command line parameter is a pain at the moment. There is also no user feedback, if wrong parameters for example REMOTE_LIMIT_API are passed.

Currently the setters of `BaseIotaConfig` do the validation. I guess you want to use Jcommander validators instead.
Anyhows, the only motivation I see is to remove Jackson as a dependency which is a fine motivation.

Just a question,
If we use Jcommander `@syntax` we will have to change the INI format?
If so we need @jakubcech agreement on it.

Personally, I don't care much about the INI format but it will break backwards compatibility.
Not a big deal, operators can adapt in my opinion.

**Comment by legacycode** [comment]
> If we use Jcommander @syntax we will have to change the INI format?
This is correct. We will break backward compatibility. But at the moment we have different keys for the same properties, eg:

ini: ZMQ_ENABLED=true -> command line: --zmq-enabled

This is only one example, but it is confusing and needs to be documented as well. Another possibility is to move the Jackson @JsonProperty things into a separate entity class and forward to the BaseIotaConfig implementation.

I suggest to drop the old ini format. For me it was confusing when setting up my iota node and, because these settings where overriden by command line parameters.

I suggest to completly drop the ini format. I don't think it is a big deal for iri node operators. On my node i use a alpine docker container with command line parameters :-)
